extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -25,38 +25,103 @@ import {EventType, dispatch} from './events';
 import {Services} from '../../../src/services';
 import {addAttributesToElement, closest, matches} from '../../../src/dom';
 import {createShadowRootWithStyle, getSourceOriginForElement} from './utils';
-import {dev, devAssert, user} from '../../../src/log';
+import {dev, devAssert, user, userAssert} from '../../../src/log';
 import {dict} from '../../../src/utils/object';
+import {findIndex} from '../../../src/utils/array';
 import {getAmpdoc} from '../../../src/service';
 import {htmlFor, htmlRefs} from '../../../src/static-template';
 import {isProtocolValid, parseUrlDeprecated} from '../../../src/url';
 import {resetStyles, setImportantStyles, toggle} from '../../../src/style';
 
 /**
- * Enum of elements that can be expanded.
+ * Action icons to be placed in tooltip.
  * @enum {string}
+ * @private
+ */
+const ActionIcon = {
+  LAUNCH: 'i-amphtml-tooltip-action-icon-launch',
+  EXPAND: 'i-amphtml-tooltip-action-icon-expand',
+};
+
+/**
+ * Components that can be expanded.
+ * @const {!Object}
+ * @private
  */
 const EXPANDABLE_COMPONENTS = {
-  TWITTER: 'amp-twitter',
+  TWITTER: {
```

Comment:
As seen offline, could you please look into if we can reuse the ones we already have for the share icons? That'd save some bytes!